service/directory-upload.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+usage="
+usage: $(basename "$0") command [-h] [-s] [-d] arguments...
+Uploads Measure and Library resources to the measure repository service.
+Options:
+    -h
+        Displays help menu.
+    -s [server baseUrl]
+        Specifies the base URL of the FHIR server to access.
+    -d [path]
+        Provides directory or file path to parse for upload.
+"
+
+while getopts ':hs:d:' option;
+do
+  case "$option" in
+    h)
+       echo -e "$usage"
+       exit 0
+       ;;
+    s)
+       server=$OPTARG
+       ;;
+    d)
+       directory_path=$OPTARG
+       ;;
+
+   \?) printf "illegal option: -%s\n" "$OPTARG" 1>&2
+       echo "$usage" 1>&2
+       exit 1
+       ;;
+    : )
+      echo "Invalid option: $OPTARG requires an argument" 1>&2
+      ;;
+  esac
+done
+
+if [[ $directory_path == "" ]] ; then
+  echo No directory path provided. Provide directory path via the '-d' flag.
+  exit 1
+fi
+
+if [[ $server == "" ]] ; then
+  echo No server URL provided. Provide server URL via the '-s' flag.
+  exit 1
+fi
+
+echo Using Server URL: $server and directory path: $directory_path
+
+upload_bundle() {
+  echo "Uploading resources for bundle $1"
+    curl_command="curl -X POST -H 'Content-Type: application/json+fhir' -d @\"$1\" $server -o /dev/null"
```

Comment:
Something I noticed that I believe solves this problem are the `logger.debug` statements in `baseUtils.ts`...now that we have a script that does a ton of requests, perhaps we should modify these? When I comment them out, I no longer get my error and I can better see the requests that are being made.

Reply:
With an ENOBUFS error, it seems like the buffer has grown too large. Eliminating some of the logging then may have limited the buffer growth and allowed this to be successful. This is something where handling may differ from system to system because of differences in resources available (memory) which could have been the cause of you seeing the error, but me being unable to reproduce it. We may be able to do some reorganization in the code, queueing, or maybe just eliminating extraneous logging to help prevent it coming up for you (and potentially on other systems).